lib/features/academic_calendar_greeting/model/weekday_enum.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+@JsonEnum()
+enum WeekdayEnum {
+  mon,
+  tue,
+  wed,
+  thu,
+  fri,
+  sat,
+  sun;
+
+  static WeekdayEnum fromJson(String json) => values.byName(json.toLowerCase());
+
+  static WeekdayEnum fromDateTime(DateTime dateTime) {
```

Comment:
I wonder if we here could swap this large switch statement with map: 


`  const weekdayMap = {`
 `  DateTime.monday: WeekdayEnum.mon,`
 `  DateTime.tuesday: WeekdayEnum.tue,....`

  and later just: 
  
`  return weekdayMap[dateTime.weekday] ?? WeekdayEnum.mon;`
}
But still, just matter of preference. Exact strategy we used with icons and assets.

Reply:
Sure, it may even have a slight better performance.

Reply:
@mikolaj-jalocha

Reply:
Yeah, I didn't notice. I agree is's good to leave it as it is.